### PR TITLE
Update Service Bus README with samples.

### PR DESCRIPTION
**Pull request by conniey**
* Updates samples README to match other languages and Event Hubs one.
* Cleans up code in samples so it adheres to guidelines
* Removes duplicated scenarios
* Adds sample for sending/receiving from a session.

### CODE REVIEW

File: sdk/servicebus/azure-messaging-servicebus/src/samples/README.md [hunk 1]
```diff
@@ -9,48 +9,62 @@ urlFragment: servicebus-samples
...
+
+### Obtaining an Event Hub instance connection string
+
+Most of the samples authorize with Service Bus using a connection string generated for that Service Bus namespace. The
+connection string value can be obtained by:
+
+1. Going to your Service Bus namespace in Azure Portal.
+1. Go to "Shared access policies" for your Service Bus namespace.
+1. Click on the "RootManageSharedAccessKey" policy.
+1. Copying the connection string from the policy's properties.
```
Comment (hemanttanwar) [inline sdk/servicebus/azure-messaging-servicebus/src/samples/README.md:29]: Why they are all `1` ?

File: sdk/servicebus/azure-messaging-servicebus/src/samples/README.md [hunk 2]
```diff
@@ -9,48 +9,62 @@ urlFragment: servicebus-samples
...
 
 ## Examples
-The following sections provide several code snippets covering some of the most common service tasks, including:
+
+- [Send messages][SendMessageAsyncSample]
+- [Send messages using Azure Identity][SendMessageWithAzureIdentityAsyncSample]
+- [Send message batches synchronously][SendMessageBatchSyncSample]
+- [Schedule and cancel a message][MessageCancelScheduleAsyncSample]
+- [Peek at a message][PeekMessageAsyncSample]
```
Comment (hemanttanwar) [inline sdk/servicebus/azure-messaging-servicebus/src/samples/README.md:37]: May be we can bring peak in bottom as every study shows that it is not used as much in prod.

File: sdk/servicebus/azure-messaging-servicebus/src/samples/README.md (hunk 1 above)
Comment (conniey) [reply sdk/servicebus/azure-messaging-servicebus/src/samples/README.md:29]: Because markdown will automatically number them.

Comment (conniey) [reply sdk/servicebus/azure-messaging-servicebus/src/samples/README.md:29]: It's part of the protocol so you don't have to renumber them manually.
https://github.com/conniey/azure-sdk-for-java/tree/add-readme/sdk/servicebus/azure-messaging-servicebus/src/samples

**Pull request merged**